pyrallis/utils.py

```diff
@@ -572,6 +574,60 @@ def deflatten(d: Dict[str, Any], sep: str = '.'):
     return deflat_d
 
 
+def parse_string(s):
+    if Config.get_config_type() == ConfigType.YAML:
+        import yaml
+        return yaml.safe_load(s)
+    elif Config.get_config_type() == ConfigType.JSON:
+        import json
+        try:
+            return json.loads(s)
+        except json.decoder.JSONDecodeError:
+            return s # No parsing to be done, simply return value
+    elif Config.get_config_type() == ConfigType.TOML:
+        import toml
+        try:
+            return toml.loads(f'val = {s}')['val']
+        except toml.decoder.TomlDecodeError:
+            return s
+    else:
+        raise PyrallisException('Unsupported config type')
+
+
+def load_config(stream):
+    if Config.get_config_type() == ConfigType.YAML:
+        import yaml
+        return yaml.full_load(stream)
+    elif Config.get_config_type() == ConfigType.JSON:
+        import json
+        return json.load(stream)
+    elif Config.get_config_type() == ConfigType.TOML:
+        import toml
+        return toml.load(stream)
+    else:
+        raise PyrallisException('Unsupported config type')
+
+
+def save_config(d, stream=None, **kwargs):
+    if Config.get_config_type() == ConfigType.YAML:
+        import yaml
+        return yaml.dump(d, stream, **kwargs)
+    elif Config.get_config_type() == ConfigType.JSON:
+        import json
+        if stream is None:
```

Comment:
`save_config` is a command, not a query. I think it shouldn't support both cases (saving and returning). Why not using an API similar to json's and toml's `dump`/`dumps`. Also this is duplicated between json and toml so maybe extract duplication...

Reply:
I'm actually following the `yaml.dump` behavior which supports both cases 

> yaml.dump accepts the second optional argument, which must be an open text or binary file. In this case, yaml.dump will write the produced YAML document into the file. Otherwise, yaml.dump returns the produced document.